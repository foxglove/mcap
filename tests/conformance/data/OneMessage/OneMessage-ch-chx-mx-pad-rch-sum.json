{
  "records": [
    {"type": "Header", "fields": [["library", ""], ["profile", ""]]},
    {
      "type": "Schema",
      "fields": [
        ["data", ["4", "5", "6"]],
        ["encoding", "c"],
        ["id", "1"],
        ["name", "Example"]
      ]
    },
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "Message",
      "fields": [
        ["channel_id", "1"],
        ["data", ["1", "2", "3"]],
        ["log_time", "2"],
        ["publish_time", "1"],
        ["sequence", "10"]
      ]
    },
    {"type": "DataEnd", "fields": [["data_section_crc", "670025892"]]},
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "ChunkIndex",
      "fields": [
        ["chunk_length", "164"],
        ["chunk_start_offset", "28"],
        ["compressed_size", "115"],
        ["compression", ""],
        ["message_end_time", "2"],
        ["message_index_length", "34"],
        ["message_index_offsets", {"1": "192"}],
        ["message_start_time", "2"],
        ["uncompressed_size", "115"]
      ]
    },
    {
      "type": "SummaryOffset",
      "fields": [
        ["group_length", "50"],
        ["group_opcode", "4"],
        ["group_start", "242"]
      ]
    },
    {
      "type": "SummaryOffset",
      "fields": [
        ["group_length", "86"],
        ["group_opcode", "8"],
        ["group_start", "292"]
      ]
    },
    {
      "type": "Footer",
      "fields": [
        ["summary_crc", "1257098038"],
        ["summary_offset_start", "378"],
        ["summary_start", "242"]
      ]
    }
  ],
  "meta": {"variant": {"features": ["ch", "mx", "rch", "chx", "sum", "pad"]}}
}
