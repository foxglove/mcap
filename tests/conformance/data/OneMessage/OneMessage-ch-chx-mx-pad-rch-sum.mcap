version https://git-lfs.github.com/spec/v1
oid sha256:22185e59d9b52f8e2cb66962eb46d65356114c73a5577a38c03f66e996733f6a
size 473
