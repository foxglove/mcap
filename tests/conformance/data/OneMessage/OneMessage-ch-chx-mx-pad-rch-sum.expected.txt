Header library="" profile=""
Schema id=1 schema=b schema_encoding=c schema_name=Example
ChannelInfo id=1 message_encoding=a metadata={foo=bar} schema_id=1 topic=example
Message channel_id=1 log_time=2 message_data=<010203> publish_time=1 sequence=10
MessageIndex channel_id=1 records={2=79}
ChannelInfo id=1 message_encoding=a metadata={foo=bar} schema_id=1 topic=example
ChunkIndex chunk_length=160 chunk_start_offset=28 compressed_size=113 compression="" end_time=2 message_index_length=34 message_index_offsets={1=190} start_time=2 uncompressed_size=113
SummaryOffset group_length=50 group_opcode=4 group_start=240
SummaryOffset group_length=0 group_opcode=13 group_start=290
SummaryOffset group_length=0 group_opcode=10 group_start=290
SummaryOffset group_length=90 group_opcode=8 group_start=290
Footer summary_crc=0 summary_offset_start=380 summary_start=240
